pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -1122,6 +1130,15 @@ func (b *MasqueradeBindMechanism) createNatRulesUsingNftables(proto iptables.Pro
 		if port.Protocol == "" {
 			port.Protocol = "tcp"
 		}
+		err = Handler.NftablesAppendRule(proto, "nat", "KUBEVIRT_POSTINBOUND",
```

Comment:
Shouldn't this rule be added outside the ports loop (only if no ports were declared on the vm)?
AFAIU the [scenario](https://github.com/istio/istio/issues/29603). If the port is declared on the pod container (in our case the virt-launcher compute container inherits ports declared on the vm/i) then 127.0.0.1 is used. Otherwise, 127.0.0.6 is used. In our case, we want to pass to the vm only ports that are declared on the vm or any port if nothing is declared. So from my understanding we should add the 127.0.0.6 rule only in case no ports are declared on the vm and it should forward any port.

Reply:
> AFAIU the scenario. If the port is declared on the pod container (in our case the virt-launcher compute container inherits ports declared on the vm/i) then 127.0.0.1 is used. Otherwise, 127.0.0.6 is used. In our case, we want to pass to the vm only ports that are declared on the vm or any port if nothing is declared.

It dosn't depend on what ports are specified in our VMI/virt-launcher pod, it is the ports that are specified in an associated k8s service resource that makes the difference. If you look at this comment in that issue https://github.com/istio/istio/issues/29603#issuecomment-746878267
You'll see this comment describes the following scenario:

```
apiVersion: v1
kind: Service
metadata:
  name: vmi-cirros
spec:
  selector:
    app: vmi-cirros
  ports:
    - port: 8080
    
Let's assume VMI with the following interface:
      interfaces:
        - name: default
          masquerade: {}
          ports:
            - port: 8080
            - port: 8081


curl <virt-launcher-pod_IP>:8080 => Envoy proxies the packet from 127.0.0.1 to 127.0.0.1:8080
curl <virt-launcher-pod_IP>:8081 => Envoy proxies the packet from 127.0.0.6 to <virt-launcher-pod_IP>:8081
```

>  So from my understanding we should add the 127.0.0.6 rule only in case no ports are declared on the vm and it should forward any port.

We could add these two rules outside the ports loop as well, where we wouldn't match the destination port. It would fix this problem for a VM without explicit ports list too. I opted for adding rules that are as specific as possible.

Reply:
Understood, thanks for the explanation.
Since this new rule and the next one are practically the same. I would add them in a for loop (iterating over the possible loopback addresses) instead of copy pasting.

Reply:
> I would add them in a for loop

Done